### ISSUE TITLE
push support requires openssl

### ISSUE THREAD

**Issue by rillian** [Ptrskay3/axum-prometheus#42]
Hi! Thanks for publishing in a update on top of the new metrics-exporter-prometheus v0.13 release.

I notice that the new release [adds](https://github.com/metrics-rs/metrics/commit/d817f5c6f4909eeafbd9ff9ceadbf29302169bfa) an `openssl-sys` dependency through the `push-gateway` feature. This expands the code size a bit and more importantly complicates the build for us to no benefit I think, since we're using polling with external tls termination.

Would it be possible to move this under a corresponding feature in `axum-prometheus`, or if there's no way to configure gateway push through this crate just build `metrics-exporter-prometheus` without it?

**Comment by Ptrskay3** [comment]
Hi. I can make the that dependency feature optional, but it's kind of unfortunate to expose `metrics-exporter-prometheus`, but not with the default features.  If I understand correctly, eventually the root cause is in `metrics-exporter-prometheus`, requiring `openssl` support by default, and you're asking for something like this:

Cargo.toml
```toml
[dependencies]
# ...
metrics = "0.22.0"
metrics-exporter-prometheus = { version =  "0.13.0", optional =  true, default-features = false, features = ["http-listener"] }

[features]
default = ["prometheus"]
prometheus = ["metrics-exporter-prometheus"]
push-gateway = ["metrics-exporter-prometheus/push-gateway"]
```

Am I right?

**Comment by rillian** [comment]
Yes, that's the sort of thing I was thinking of.

I hadn't thought about closing it off to re-exports. Is `metrics_exporter_prometheus::PrometheusBuilder` actually available without referencing that crate directly? If not it seems like you could just disable the feature, and leave `push-gateway` for a future change to add more direct push support.

**Comment by rillian** [comment]
Nevermind, the re-export is [here](https://github.com/Ptrskay3/axum-prometheus/blob/0.6.0/src/lib.rs#L209-L211) so it's worth repeating the feature here for those using it.

**Comment by Ptrskay3** [comment]
In #44 I ended up just disabling the feature, and leaving a note in the README. In `axum_prometheus` we don't need that feature on as far as I can tell. If someone really wants it, they can turn it back on. I think I'll merge that PR soon, if that suits your needs.

**Comment by rillian** [comment]
Yes, #44 addresses the issue I had. Thanks for taking a look!

**Comment by Ptrskay3** [comment]
Just released `0.6.1` that fixes your issue. Thanks for pointing it out.
